data/transactions/logic/eval.go

```diff
@@ -247,10 +247,21 @@ type LedgerForLogic interface {
 	Counter() uint64
 }
 
-// boxRef is the "hydrated" form of a BoxRef - it has the actual app id, not an index
-type boxRef struct {
-	app  basics.AppIndex
```

Comment:
The confusion here is because we sometimes call app/asset IDs app/asset indexes. `basics.AppIndex` is an app ID.

The "not an index" part of this docstring refers to the fact that this is an app ID, _not_ an index into the foreign app array, which is what `transaction.BoxRef` uses.